File: raspberry/windows-iot/MetecDriver/SingleCellExample/StartupTask.cpp
#include "pch.h"
#define CELL_COUNT 20
#define ON         22
#define DIN        4
#define STROBE     17
#define CLK        27
#define DOUT       18

using namespace MetecDriverLibrary;
using namespace SingleCellExample;
using namespace Windows::ApplicationModel::Background;

#define PATTERN_LENGTH 11
uint16_t pattern[PATTERN_LENGTH] = {
	0x2813, 0x2811, 0x2807, 0x2807, 0x2815, 0x20, 0x283a, 0x2815, 0x2817, 0x2807, 0x2819
	//     ⠓       ⠑      ⠇       ⠇       ⠕             ⠺       ⠕      ⠗       ⠇      ⠙
	//     H       E      L       L        O             W       O       R      L       D
};

MetecDriver metecDriver(CELL_COUNT, ON, DIN, STROBE, CLK, DOUT);

void StartupTask::Run(IBackgroundTaskInstance^ taskInstance)
{
	metecDriver.init();
	for (uint8_t i = 0; i < PATTERN_LENGTH; i++) {
		metecDriver.writeSingleCell(CELL_COUNT - 1 - i, pattern[i]);
	}
}

File: raspberry/windows-iot/MetecDriver/ExampleSingleCell/StartupTask.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Net.Http;
using Windows.ApplicationModel.Background;

using MetecDriverLibrary;

namespace ExampleSingleCell
{
    public sealed class StartupTask : IBackgroundTask
    {
        const byte CELL_COUNT = 20;
        const byte ON = 22;
        const byte DIN = 4;
        const byte STROBE = 17;
        const byte CLK = 27;
        const byte DOUT = 18;

        MetecDriver metecDriver = new MetecDriver(CELL_COUNT, ON, DIN, STROBE, CLK, DOUT);

        ushort[] pattern = new ushort[] {
            0x2813, 0x2811, 0x2807, 0x2807, 0x2815, 0x20, 0x283a, 0x2815, 0x2817, 0x2807, 0x2819 };
        //     ⠓       ⠑      ⠇       ⠇       ⠕             ⠺       ⠕      ⠗       ⠇      ⠙
        //     H       E       L       L        O             W       O       R       L       D

        public void Run(IBackgroundTaskInstance taskInstance)
        {
            metecDriver.init();
            for (int i = 0; i < pattern.Length; i++)
            {
                metecDriver.writeSingleCell((byte)(CELL_COUNT - 1 - i), pattern[i]);
            }
        }
    }
}
